platforms/chibios/drivers/i2c_master.c

```diff
@@ -107,16 +107,14 @@ static const I2CConfig i2cconfig = {
 #endif
 };
 
-static i2c_status_t chibios_to_qmk(const msg_t* status) {
-    switch (*status) {
-        case I2C_NO_ERROR:
-            return I2C_STATUS_SUCCESS;
-        case I2C_TIMEOUT:
-            return I2C_STATUS_TIMEOUT;
-        // I2C_BUS_ERROR, I2C_ARBITRATION_LOST, I2C_ACK_FAILURE, I2C_OVERRUN, I2C_PEC_ERROR, I2C_SMB_ALERT
-        default:
-            return I2C_STATUS_ERROR;
+static i2c_status_t chibios_to_qmk(const msg_t status) {
+    if (status == I2C_NO_ERROR) {
+        return I2C_STATUS_SUCCESS;
     }
+
+    i2c_stop();
```

Comment:
I really don't like this here. `chibios_to_qmk()` no longer just converts between error codes.

Reply:
Good point I'll rename it or split that part out

Reply:
As discussed on Discord, `chibios_to_qmk` is now `i2c_epilogue` and does error handling + conversion.

Reply:
Note that this function was and still is buggy — according to the [ChibiOS docs](http://chibiforge.org/doc/20.3/full_rm/group___i2_c.html), the `msg_t` value returned by I2C driver functions like `i2cMasterTransmitTimeout()` is actually one of `MSG_OK`, `MSG_RESET`, `MSG_TIMEOUT`, not the I2C status. The actual I2C error code like `I2C_TIMEOUT`, `I2C_BUS_ERROR`, … is returned by `i2cGetErrors(i2cp)`, which must be called only when the I2C driver function returned `MSG_RESET`.

Also the value returned by `i2cGetErrors()` is a bit mask that can have multiple error bits set at once, so it needs to be tested with something like `status & I2C_TIMEOUT`.